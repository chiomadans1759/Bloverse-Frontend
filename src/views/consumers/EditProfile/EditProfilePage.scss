@import 'sass-rem';

.logo {
  height: rem(40px);
  padding-left: rem(20px);
}
.top {
    background: white;
    padding: rem(10px);
}
.top-left-container {
  margin-left: 5%;
}
.top-left-container > a {
  color: #000000 !important;
}
.top-left-container > a > svg {
  vertical-align: middle;
}
.top-left-container > a > span {
  padding-left: rem(10px);
}
.content{
  h3{
    font-size: rem(27px);
    color: #353535;
    font-weight: 900;
    margin-bottom: rem(4px);
  }
  input{
    margin-bottom: 3%;
  }
  .file_share {
    height: rem(50px);
    background: white;
    border: rem(1px) solid #096DD9;
    border-radius: rem(9px);
    margin-top: -13%;
    svg{
      color: #096DD9;
      font-size: rem(31px);
      display: block;
      text-align: center;
      margin: 0 auto;
      padding-top: 16%;
  }
    input {
      position: absolute;
      width: 77%;
      height: 100%;
      opacity: 0;
    }
  }
}
.edit__profile--select{
  -webkit-appearance:none;
  border-radius: rem(4px) 0 0 rem(4px);
  border-right: none;
}
