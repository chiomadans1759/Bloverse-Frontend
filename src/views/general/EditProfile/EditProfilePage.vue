<template>
  <div class="edit-container">
    <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" customClass="back-nav"/>
    <div class="content">
      <div class="container">
        <div class="row">
          <div class="col-md-4"></div>
          <div class="col-12 col-md-4 col-sm-4 col-lg-4">
            <br>
            <br>

            <h3>Edit Profile</h3>

            <br>
            <br>
            <br>
            <form action>
              <div class="row">
                <div class="col-3 col-md-3 col-sm-3 col-lg-3 pr-0">
                  <div class="file_share">
                    <input
                      type="file"
                      name="profileImage"
                      id="profile-image"
                      v-on:change="onChange"
                    >
                    <i class="fal fa-plus"></i>
                  </div>
                </div>
                <div class="col-9 col-md-9 col-sm-9 col-lg-9">
                  <input
                    type="text"
                    class="form-control"
                    placeholder="User Name"
                    name="username"
                    id="username"
                    v-model="user.username"
                    v-on:change="onChange"
                  >
                </div>
              </div>
              <input
                type="text"
                class="form-control"
                placeholder="First Name"
                name="firstName"
                id="firstName"
                v-model="user.firstname"
                v-on:change="onChange"
              >
              <input
                type="text"
                class="form-control"
                placeholder="Last Name"
                name="lastName"
                id="lastName"
                v-model="user.lastname"
                v-on:change="onChange"
              >
              <div class="row">
                <div class="col-md-3 col-3 pr-0">
                  <select
                    name="countrycode"
                    id="countrycode"
                    v-model="user.countrycode"
                    class="form-control edit__profile--select"
                    v-on:change="onChange"
                  >
                    <option value>+234</option>
                    <option value="+234">+234</option>
                    <option value="+234">+234</option>
                    <option value="+234">+234</option>
                    <option value="+234">+234</option>
                    <option value="+234">+234</option>
                  </select>
                </div>
                <div class="col-md-9 col-9 pl-0">
                  <input
                    type="number"
                    class="form-control"
                    placeholder="000 000 0000"
                    name="phone"
                    id="phone"
                    v-model="user.phone"
                    v-on:change="onChange"
                  >
                </div>
              </div>
              <input
                type="email"
                class="form-control"
                placeholder="Email Adddress"
                name="email"
                id="email"
                v-model="user.email"
                v-on:change="onChange"
              >
              <input
                type="text"
                name="location"
                v-model="user.location"
                id="location"
                class="form-control"
                placeholder="Lagos Nigeria"
                v-on:change="onChange"
              >
              <textarea
                name="bio"
                id="bio"
                v-model="user.bio"
                cols="30"
                rows="4"
                class="form-control"
                v-on:change="onChange"
              ></textarea>
              <br>
              <span class="d-flex justify-content-end span-text">0&#x3C;124</span>
              <br>
              <button type="submit" class="btn btn-primary btn-block edit-btn">Save</button>
            </form>
          </div>
          <div class="col-md-4"></div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Navbar from "@/components/Navbar/Navbar.vue";

export default {
  name: "EditProfile",
  data() {
    return {
      user: {
        firstname: "",
        lastname: "",
        bio: "Bio",
        phone: "",
        location: "",
        countrycode: "",
        username: "",
        profileimage: ""
      },
      showNavigations: false,
      showBackArrow: true
    };
  },
  components: { Navbar },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  }
};
</script>

<style lang="scss" scoped="true">
@import "./EditProfilePage.scss";
</style>
