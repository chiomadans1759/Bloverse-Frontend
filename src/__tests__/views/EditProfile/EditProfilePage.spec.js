import { mount, RouterLinkStub } from '@vue/test-utils'
import EditProfilePage from '@/views/general/EditProfile/EditProfilePage.vue';

describe('EditProfile Page', () => {
  it('should render correctly', () => {

    const wrapper = mount(EditProfilePage, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should update the page state when input is changed', () => {
    const wrapper = mount(EditProfilePage, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const firstNameInput = wrapper.find('#firstName');
    firstNameInput.element.value = 'firstname';
    firstNameInput.trigger('change');

    const lastNameInput = wrapper.find('#lastName');
    lastNameInput.element.value = 'lastname';
    lastNameInput.trigger('change');

    const emailInput = wrapper.find('#email');
    emailInput.element.value = 'test@email.com';
    emailInput.trigger('change');

    const bioInput = wrapper.find('#bio');
    bioInput.element.value = "i am a boy";
    bioInput.trigger('change');

    const countryCodeInput = wrapper.find('#countrycode');
    countryCodeInput.element.value = "+234";
    countryCodeInput.trigger('change');

    const userNameInput = wrapper.find('#username');
    userNameInput.element.value = "test333";
    userNameInput.trigger('change');

    const phoneInput = wrapper.find('#phone');
    phoneInput.element.value = "080123456";
    phoneInput.trigger('change');


    const locationInput = wrapper.find('#location');
    locationInput.element.value = "nigeria";
    locationInput.trigger('change');

    expect(wrapper.vm.firstName).toBe('firstname');
    expect(wrapper.vm.lastName).toBe('lastname');
    expect(wrapper.vm.email).toBe('test@email.com');
    expect(wrapper.vm.bio).toBe('i am a boy');
    expect(wrapper.vm.countrycode).toBe('+234');
    expect(wrapper.vm.username).toBe('test333');
    expect(wrapper.vm.phone).toBe('080123456');
    expect(wrapper.vm.location).toBe('nigeria');
  });
});
