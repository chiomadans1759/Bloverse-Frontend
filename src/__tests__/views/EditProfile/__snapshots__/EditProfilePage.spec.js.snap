// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile Page should render correctly 1`] = `
<div
  class="edit-container"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5"
    >
      <div
        class="container-fluid back-nav"
      >
        <a
          class="backlink mr-4"
          href="/settings"
        >
          <i
            class="fal fa-long-arrow-left fa-2x back-icon"
          />
           
          <span
            class="back-text"
          >
            Back
          </span>
        </a>
         
        <a
          class="router-link"
        >
          <img
            class="nav-logo"
            src="@/assets/Bloverse.svg"
          />
           
          <!---->
        </a>
         
        <!---->
         
        <!---->
      </div>
    </nav>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Bug
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Comment
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Other
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                
              Close
            
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </header>
   
  <div
    class="content"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-4"
        />
         
        <div
          class="col-12 col-md-4 col-sm-4 col-lg-4"
        >
          <br />
           
          <br />
           
          <h3>
            Edit Profile
          </h3>
           
          <br />
           
          <br />
           
          <br />
           
          <form
            action=""
          >
            <div
              class="row"
            >
              <div
                class="col-3 col-md-3 col-sm-3 col-lg-3 pr-0"
              >
                <div
                  class="file_share"
                >
                  <input
                    id="profile-image"
                    name="profileImage"
                    type="file"
                  />
                   
                  <i
                    class="fal fa-plus"
                  />
                </div>
              </div>
               
              <div
                class="col-9 col-md-9 col-sm-9 col-lg-9"
              >
                <input
                  class="form-control"
                  id="username"
                  name="username"
                  placeholder="User Name"
                  type="text"
                />
              </div>
            </div>
             
            <input
              class="form-control"
              id="firstName"
              name="firstName"
              placeholder="First Name"
              type="text"
            />
             
            <input
              class="form-control"
              id="lastName"
              name="lastName"
              placeholder="Last Name"
              type="text"
            />
             
            <div
              class="row"
            >
              <div
                class="col-md-3 col-3 pr-0"
              >
                <select
                  class="form-control edit__profile--select"
                  id="countrycode"
                  name="countrycode"
                >
                  <option
                    value=""
                  >
                    +234
                  </option>
                   
                  <option
                    value="+234"
                  >
                    +234
                  </option>
                   
                  <option
                    value="+234"
                  >
                    +234
                  </option>
                   
                  <option
                    value="+234"
                  >
                    +234
                  </option>
                   
                  <option
                    value="+234"
                  >
                    +234
                  </option>
                   
                  <option
                    value="+234"
                  >
                    +234
                  </option>
                </select>
              </div>
               
              <div
                class="col-md-9 col-9 pl-0"
              >
                <input
                  class="form-control"
                  id="phone"
                  name="phone"
                  placeholder="000 000 0000"
                  type="number"
                />
              </div>
            </div>
             
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="Email Adddress"
              type="email"
            />
             
            <input
              class="form-control"
              id="location"
              name="location"
              placeholder="Lagos Nigeria"
              type="text"
            />
             
            <textarea
              class="form-control"
              cols="30"
              id="bio"
              name="bio"
              rows="4"
            />
             
            <br />
             
            <span
              class="d-flex justify-content-end span-text"
            >
              0&lt;124
            </span>
             
            <br />
             
            <button
              class="btn btn-primary btn-block edit-btn"
              type="submit"
            >
              Save
            </button>
          </form>
        </div>
         
        <div
          class="col-md-4"
        />
      </div>
    </div>
  </div>
</div>
`;
